bitar/src/api/compress.rs

```diff
@@ -0,0 +1,269 @@
+use std::collections::HashMap;
+use std::default::Default;
+use std::error;
+use std::fmt;
+use std::path::PathBuf;
+
+use blake2::{Blake2b512, Digest};
+use futures_util::{future, StreamExt};
+use tokio::fs;
+use tokio::io;
+use tokio::io::AsyncSeekExt;
+use tokio::io::{AsyncRead, AsyncWrite, AsyncWriteExt};
+
+use crate::chunk_dictionary;
+use crate::chunker;
+use crate::Compression;
+use crate::HashSum;
+
+pub const PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Options for the `create_archive` function
+#[derive(Clone, Debug)]
+pub struct CreateArchiveOptions {
+    /// The configuration to use when creating a chunk stream from the input
+    pub chunker_config: chunker::Config,
+
+    /// Number of parallel buffers to use when manipulating chunks
+    pub num_chunk_buffers: usize,
+
+    /// The length that the source hash should be truncated to for the output
+    pub source_hash_length: usize,
+
+    /// A temporary file is used to write intermediate chunk data. Setting this
+    /// option forces this file to be used instead of a randomly generated one
+    pub temporary_file_override: Option<PathBuf>,
+
+    /// The type of compression to use when compressing a chunk
+    pub compression: Option<Compression>,
+}
+
+impl Default for CreateArchiveOptions {
+    fn default() -> CreateArchiveOptions {
+        let num_buffers = match num_cpus::get() {
+            0 | 1 => 1,
+            n => n * 2,
+        };
+
+        CreateArchiveOptions {
+            chunker_config: chunker::Config::FixedSize(64),
+            num_chunk_buffers: num_buffers,
+            source_hash_length: HashSum::MAX_LEN,
+            temporary_file_override: None,
+            compression: None,
+        }
```

Comment:
These defaults and the defaults from bita (cli) does not match. And I'm thinking that they probably should.
And I would prefer if these were changed to match the default of bita.

As a side note... Wouldn't chunking with a fixed size of 64 bytes and a hash length of 64 bytes yield an archive with a dictionary which has the same size as the source file? Hence when cloning one would have to fetch both the diff of chunks but also a dictionary big as the original file. Then better not to use bitar at all. :slightly_smiling_face:

Reply:
I updated the defaults to match those of bita, which is rullsum + brotli level 6 and 64-byte chunk hashes